lib/container_scheduler.rb

```diff
@@ -1,76 +1,49 @@
 class ContainerScheduler
-  def initialize
+  def initialize(limit_mem_threshold: nil, limit_n_containers: nil, limit_n_stateful_containers: nil)
+    @limit_mem_threshold = limit_mem_threshold
+    @limit_n_containers = limit_n_containers
+    @limit_n_stateful_containers = limit_n_stateful_containers
   end
 
   def schedule
-    if ENV['SCHEDULER_TYPE'] == 'MEMORY'
-      p 'Scheduling container based on node memory'
-    else
-      p 'Scheduling container based on node container numbers'
+    n_containers = 0
```

Comment:
why delete all the previous method, why not provide options?

Reply:
Previous available options:
- Limit based-on memory.
- Limit based-on container numbers.

In the current implementation, all of these limitations can be applied or not by specifying or not specifying the environment variable. For example, if:
- I only want the memory limit, I can just define `SCHEDULER_MEMORY_THRESHOLD`.
- I only want the container count limit, I can just define ..., wait, I just saw that there's no container-limit feature in the old implementation.

Well, I deleted the previous implementation because I thought that my new implementation will unify the previous implementation. Do you have any intention of this? E.g. I want the previous (sorting) method to be still available.

Reply:
intention: possibility to have pluggable implementation in the future. So people can just add more options by writing their own.

Reply:
@giosakti 
I've refactored it into functions. So it'll be easier to add more limits. Will it solve this?

Reply:
Ping @giosakti

It's pluggable here. If we need to add more limits, we just need to add more line here:
https://github.com/pathfinder-cm/pathfinder-mono/pull/51/files#diff-530b10a285d4bbaa401cffeb11b3fdd7R35-R37

Then the function will alter the query. Does it help? Thank you.